volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -116,3 +165,369 @@ def lookup_module_address(
+        except exceptions.InvalidAddressException:
+            return None
+
+        start = module.get_module_base() & address_mask
+
+        end = start + module.get_core_size()
+
+        return Modules.ModuleInfo(module.vol.offset, mod_name, start, end)
+
+    @staticmethod
+    def get_kernel_module_info(
+        context: interfaces.context.ContextInterface,
+        kernel_module_name: str,
+    ) -> ModuleInfo:
+        """
+        Returns a ModuleInfo instance that encodes the kernel
+        This is required to map function pointers to the kerenl executable
+        """
+        kernel = context.modules[kernel_module_name]
+
+        mask = context.layers[kernel.layer_name].address_mask
+
+        start_addr = kernel.object_from_symbol("_text")
+        start_addr = start_addr.vol.offset & mask
+
+        end_addr = kernel.object_from_symbol("_etext")
+        end_addr = end_addr.vol.offset & mask
+
+        return Modules.ModuleInfo(
+            start_addr, constants.linux.KERNEL_NAME, start_addr, end_addr
+        )
+
+    @classmethod
+    def run_modules_scanners(
+        cls,
+        context: interfaces.context.ContextInterface,
+        kernel_name: str,
+        run_hidden_modules: bool = True,
+        flatten: bool = True,
+    ) -> Dict[str, List[ModuleInfo]]:
+        """Run module scanning plugins and aggregate the results. It is designed
+        to not operate any inter-plugin results triage.
+
+        Args:
+            run_hidden_modules: specify if the hidden_modules plugin should be run
+        Returns:
+            Dictionary mapping each plugin to its corresponding result
+        """
+
+        kernel = context.modules[kernel_name]
+
+        address_mask = context.layers[kernel.layer_name].address_mask
+
+        run_results = {}
+
+        # the kernel module boundaries
+        run_results["kernel"] = [cls.get_kernel_module_info(context, kernel_name)]
+
+        # lsmod
+        run_results["lsmod"] = []
+
+        for module in cls.list_modules(context, kernel_name):
+            modinfo = cls.get_module_info_for_module(address_mask, module)
+            if modinfo:
+                run_results["lsmod"].append(modinfo)
+
+        # check_modules
+        run_results["check_modules"] = []
+
+        sysfs_modules: dict = cls.get_kset_modules(context, kernel_name)
+
+        for m_offset in sysfs_modules.values():
+            module = kernel.object(object_type="module", offset=m_offset, absolute=True)
+            modinfo = cls.get_module_info_for_module(address_mask, module)
+            if modinfo:
+                run_results["check_modules"].append(modinfo)
+
+        # hidden_modules
+        if run_hidden_modules:
+            known_modules_addresses = set(
+                context.layers[kernel.layer_name].canonicalize(modinfo.start)
+                for modinfo in run_results["kernel"]
+                + run_results["lsmod"]
+                + run_results["check_modules"]
```

Comment:
```suggestion
                for modinfo in cls.flatten_run_modules_results(run_results)
```
This should probably be driven by the contents of the list, rather than hard coding the available tests, which looks to be what flatten_run_modules_results does?

Reply:
Yes this will be a part of: https://github.com/volatilityfoundation/volatility3/issues/1699